src/SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure/Events/FinalisedOnProgrammeLearningPaymentEventHandler.cs

```diff
@@ -0,0 +1,18 @@
+﻿using NServiceBus;
+using SFA.DAS.Funding.ApprenticeshipPayments.Types;
+using System.Collections.Concurrent;
+
+namespace SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure.Events;
+
+public class FinalisedOnProgrammeLearningPaymentEventHandler: IHandleMessages<FinalisedOnProgammeLearningPaymentEvent>
+{
+    public static ConcurrentBag<FinalisedOnProgammeLearningPaymentEvent> ReceivedEvents { get; } = new();
+
+    public Task Handle(FinalisedOnProgammeLearningPaymentEvent message, IMessageHandlerContext context)
+    {
+        ReceivedEvents.Add(message);
+        return Task.CompletedTask;
+    }
+
+    public void ClearEvent() => ReceivedEvents.Clear();
```

Comment:
I don't think you are using this method atm, the calling code just calls ReceivedEvents.Clear() directly